src/solver.h

```diff
@@ -37,6 +43,7 @@ struct Solver {
     const double dt = 1.;
     state_t state;
     integration_function_t ode;
+    observer_t observer;
```

Comment:
You might need to add an additional control parameter to set a max step size equal to the delay length. Otherwise when you go back to find the interval that bounds `t - tau` you won't find it because it's part of the current step when `h > tau`